tests/rec/test_gender.py

```diff
@@ -0,0 +1,138 @@
+        http_requests.append(last_request)
+
+        params = last_request.querystring
+        name = params['name'][0].lower()
+
+        if name == 'error':
+            return 502, headers, 'Bad Gateway'
+
+        if name == 'john':
+            data = {
+                'gender': 'male',
+                'probability': 0.92
+            }
+        elif name == 'jane':
+            data = {
+                'gender': 'female',
+                'probability': 0.89
+            }
+        else:
+            data = {
+                'gender': None,
+                'probability': None
+            }
+
+        body = json.dumps(data)
+
+        return (200, headers, body)
+
+    httpretty.register_uri(httpretty.GET,
+                           GENDERIZE_API_URL,
+                           responses=[
+                               httpretty.Response(body=request_callback)
+                           ])
+
+    return http_requests
+
+
+class TestRecommendGender(TestCase):
+    """Unit tests for recommend_gender"""
+
+    def setUp(self):
+        """Initialize database with a dataset"""
+
+        self.user = get_user_model().objects.create(username='test')
+        self.ctx = SortingHatContext(self.user)
+
+        self.john_smith = api.add_identity(self.ctx,
+                                           name='John Smith',
+                                           source='scm')
+        self.jane_doe = api.add_identity(self.ctx,
+                                         name='Jane Doe',
+                                         source='scm')
+        self.smith = api.add_identity(self.ctx,
+                                      name='Smith',
+                                      source='scm')
+        self.no_name = api.add_identity(self.ctx,
+                                        email='email@example.com',
+                                        source='scm')
+
+    @httpretty.activate
+    def test_recommend_gender(self):
+        """Check if it returns a gender for valid names"""
+
+        setup_genderize_server()
+
+        uuids = [self.john_smith.uuid, self.jane_doe.uuid]
+        recs = list(recommend_gender(uuids))
+
+        self.assertEqual(len(recs), 2)
+
+        rec = recs[0]
+        self.assertEqual(rec[0], self.john_smith.uuid)
+        gender, acc = rec[1]
+        self.assertEqual(gender, 'male')
+        self.assertEqual(acc, 92)
+
+        rec = recs[1]
+        self.assertEqual(rec[0], self.jane_doe.uuid)
+        gender, acc = rec[1]
+        self.assertEqual(gender, 'female')
+        self.assertEqual(acc, 89)
+
+    @httpretty.activate
+    def test_no_name(self):
+        """Check if no recommendations are generated when an
+            individual does not have a valid name"""
+
+        setup_genderize_server()
+
+        uuids = [self.smith.uuid, self.no_name.uuid]
+        recs = list(recommend_gender(uuids))
+
+        self.assertEqual(len(recs), 0)
+
+    def test_not_found_individual(self):
```

Comment:
I think that this test should be also included in the tests for the mutations so we can check that everything goes ok.

Reply:
We finally decided this wasn't needed because during the mutation tests we mock the response, so this function is not going to fail.